### ISSUE TITLE
Give input format to date.parse

### ISSUE THREAD

**Issue by yogeshwar-b** [scriban/scriban#404]
How to give input Date Format to date.parse method ? 

I have looked at the documentation but did not find it . 

Model : 
`{"dob":"20/01/2022 08:32:48 +00:00" }`

Scriban Template : 
```
{{~
	func formatDate
		ret (date.parse $0)| date.to_string '%Y-%m-%d %H:%M:%S'
	end
~}}


DOB {{formatDate model.dob}}
```

Actual Output :

`DOB 0001-01-01 00:00:00`

Expected Output : 

`DOB 2022-01-20 08:32:48`

How to get expected output ?

Sorry if this is wrong place to ask this question.
Thanks in Advance.

**Comment by xoofx** [comment]
The function used to parse is define here:

https://github.com/scriban/scriban/blob/78de3af89c0c1bcca04bbb228df4008a077d1164/src/Scriban/Functions/DateTimeFunctions.cs#L250-L262

I haven't checked, but it could be that this doesn't handle well when the time passed as time offsets, so it might require to try to parse it multiple steps (e.g with `DateTimeOffset.TryParse` if `DateTime.TryParse` fails)

PR Welcome.

**Comment by yogeshwar-b** [comment]
I see that the date parse is working fine if the context culture is 'en-GB' in line 257 in above mentioned code.

I dont have access to the Scriban render code. Only thing i can modify is Template file , can i set culture from the template file ?

**Comment by xoofx** [comment]
> I dont have access to the Scriban render code. Only thing i can modify is Template file , can i set culture from the template file ?

You could see from the code above that it was not possible to pass a culture.

Anyway, I fixed it in 072aa23, it should be available in the upcoming`5.2.0+`

I won't do that always (to bring something that you could have tried to bring to this repo yourself), so yep, either if you see me one day, buy me a beer, or please, consider making a one-time donation through GitHub. 😉
